### DISCUSSION TITLE
Pocketbase Framework (jsvm): $http.send and gzip response

### DISCUSSION THREAD

**Discussion by jonshipman** [pocketbase/pocketbase#4176] [Q&A] [answered]
How would I go about parsing gzipped content when using $http.send?

**Comment by ganigeorgiev** [comment]
Hm, there is no helper for decompressing gzip responses at the moment.

I'll consider in the next v0.21/v0.22 release to check the `content-encoding` header and decompress it automatically before returning the response to the user.

**Reply by ganigeorgiev** [reply, marked as answer]
Actually the auto decompression may not be needed. Based on the [default Go `Transport.DisableCompression` doc](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/net/http/transport.go;l=195), if no explicit `Accept-Encoding: gzip` was sent with the request, then the body will be transparently decoded for us:

> If the Transport requests gzip on its own and gets a gzipped response, it's transparently decoded in the Response.Body.
> However, if the user explicitly requested gzip it is not automatically uncompressed.

I've tested it locally with fetching a gzip-ed json response and it returns json correctly when the request doesn't have explicit `Accept-Encoding: gzip` header.

```js
const res = $http.send({
    url:     "http://localhost:8090/test",
    method:  "GET",
    headers: {
        "content-type": "application/json",
        // explicitly requesting it will not decompress the response body
        // "Accept-Encoding": "gzip",
    },
})

console.log(JSON.stringify(res.json)) // outputs the decompressed json response
```
